app/src/main/java/com/github/se/polyfit/ui/screen/recipeRec/RecipeRecommendationMoreDetailScreen.kt

```diff
@@ -41,47 +40,50 @@ fun RecipeRecommendationMoreDetailScreen(
             title = ContextCompat.getString(context, R.string.MoreDetailRecipe),
             navigateBack = navigateBack)
       }) { padding ->
-        RecipeDetailContent(recipe, padding, recipeRecViewModel)
+        RecipeDetailContent(padding, recipeRecViewModel)
       }
 }
 
 @Composable
-fun RecipeDetailContent(
-    recipe: Recipe,
-    padding: PaddingValues,
-    recipeRecViewModel: RecipeRecommendationViewModel
-) {
+fun RecipeDetailContent(padding: PaddingValues, recipeRecViewModel: RecipeRecommendationViewModel) {
   val showIngredient by recipeRecViewModel.showIngredient.observeAsState(initial = true)
+  val selectedRecipe: Recipe? by recipeRecViewModel.selectedRecipe.observeAsState()
 
-  val context = LocalContext.current
-  LazyColumn(
-      horizontalAlignment = Alignment.CenterHorizontally,
-      modifier = Modifier.testTag("RecipeList").padding(padding)) {
-        item { RecipeCard(recipe = recipe, showTitle = false) }
-        item {
-          TitleAndToggleCard(
-              title = recipe.title,
-              button1Title = ContextCompat.getString(context, R.string.Ingredients),
-              button2Title = ContextCompat.getString(context, R.string.Recipe),
-              recipeRecViewModel)
-        }
-        item { NutriInfoCard(recipeInfo = recipe.recipeInformation) }
+  if (selectedRecipe == null) {
+    Box(modifier = Modifier.fillMaxSize().padding(padding), contentAlignment = Alignment.Center) {
+      CircularProgressIndicator(modifier = Modifier.padding(16.dp).testTag("LoadingPost"))
+    }
+    return
+  } else {
 
-        if (showIngredient) {
+    val context = LocalContext.current
+    LazyColumn(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier =
+            Modifier.testTag("RecipeList")
+                .padding(padding)
+                .padding(bottom = 16.dp) // Add bottom padding here
+        ) {
+          item { RecipeCard(recipe = selectedRecipe!!, showTitle = false) }
           item {
-            // This is done purely for testing purposes
-            Box(modifier = Modifier.fillMaxSize().testTag("IngredientList")) {
-              recipe.recipeInformation.ingredients.forEach { IngredientInfoCard(ingredient = it) }
+            TitleAndToggleCard(
+                title = selectedRecipe!!.title,
+                button1Title = ContextCompat.getString(context, R.string.Ingredients),
+                button2Title = ContextCompat.getString(context, R.string.Recipe),
+                recipeRecViewModel)
+          }
+
+          if (showIngredient) {
+            selectedRecipe?.recipeInformation?.ingredients?.forEach {
+              item { IngredientInfoCard(ingredient = it) }
+            }
+          } else {
+            selectedRecipe!!.recipeInformation.instructions?.forEachIndexed { index, step ->
```

Comment:
```suggestion
            selectedRecipe!!.recipeInformation.instructions.forEachIndexed { index, step ->
```
i think this value should always exist, no need for safe assert